app/src/features/apiClient/helpers/APIClientManager/modules/scripts/workerScripts.ts

```diff
@@ -132,10 +169,57 @@ export const responseWorkerFunction = function (e: MessageEvent) {
     return new Proxy(() => {}, handler);
   };
 
+  const recursiveJSONify = (obj: any): any => {
+    try {
+      // If it's a JSON string, try to parse it first
+      if (typeof obj === "string") {
+        try {
+          return recursiveJSONify(JSON.parse(obj));
+        } catch {
+          return obj;
+        }
+      }
+
+      if (typeof obj !== "object" || obj === null) {
+        return obj;
+      }
+
+      if (Array.isArray(obj)) {
+        return obj.map((item) => recursiveJSONify(item));
+      }
+
+      return Object.entries(obj).reduce((acc, [key, value]) => {
+        acc[key] = recursiveJSONify(value);
+        return acc;
+      }, {} as any);
+    } catch {
+      return String(obj);
+    }
+  };
+
   const sandbox = {
     rq: {
-      request,
-      response,
+      request: {
+        ...request,
+        toJSON: () => {
+          return {
+            method: request.method,
+            url: request.url,
+            body: recursiveJSONify(request.body),
+          };
```

Comment:
This doesn't ensure all the future properties are added automatically in the toJSON return type. We'll need to manually add it everytime

Reply:
```
Object.setPrototypeOf(requestFinal, {
    toJSON: function() { return JSON.parse(JSON.stringify(this)); }
});
```
Can't we do something like this?